pkg/arvo/neo/cod/std/src/fil/messenger.md

```diff
@@ -562,3 +563,273 @@ The `stem` is head-tagged with a `care:neo`. If the shrub’s dependent is liste
+
+#### kook:neo
+
+```hoon
+::  /imp/dm.hoon
+++  state  [%pro %ship]
+++  poke  (sy %dm-diff ~)
+++  kids
+  %-  some
+  :-  %y
+  %-  ~(gas by *lads:neo)
+  :~  :-  [[%.n %theirs] %.n]
+      [[%pro %message-pub] (sy %sig ~)]
+      :-  [[%.n %mine] %.n]
+      [[%pro %message-sub] (sy %sig ~)]
+  ==
+++  deps  *deps:neo
+++  form
+  ...
+```
+
+A DM shrub only stores one ship, the `@p` of whoever you're chatting with. It only has two kids: `/path/to/this/dm/theirs` and `/path/to/this/dm/mine`. It chooses the service providers at `/imp/message-pub` and `/imp/message-sub` to store the state of `/theirs` and `/mine`, while setting no constraints on the types of poke those service providers can take. In other words, it delegates the functionality to two new instances of `/imp/message-pub` and `/imp/message-sub` rather than subscribing to existing Gall agents that may or may not exist.
+
+```hoon
+$%  [%initiate partner=ship provider=pith]
+    [%invited partner=ship dm=pith]
+  ::
+    [%acked dm=pith]
+  ::
+    [%post text=@t]
+==
+```
+
+`/imp/dm` takes four pokes: `%initiate`, `%invited`, `%acked`, and `%post`.
+- `%initiate`: we initiate a DM, specifying the other ship and the service provider we want to use.
+- `%invited`: someone invites us to a DM, specifying their `@p` and the `pith` for us to send DMs to them.
+- `%acked`: acknowledge creation of a DM.
+- `%post`: send a DM.
+
+#### curb:neo
+Cells like `[%pro %ship]` and `[%pro %message-pub]` are examples of `$curb:neo`. This is a powerful type that's beyond the remit of these tutorials, but it's worth clarifying what these cells mean.
+
+```hoon
++$  curb
+  $~  [%pro %$]
+  $%  [%or p=(list curb)]
+      [%only p=stud]
+      [%rol p=stud q=curb]
+      [%not p=curb q=curb]
+      [%pro p=stud]
+      [%any ~]
+  ==
+```
+
+In all of the shrubs we've looked at in these tutorials we could replace every `%pro` curb with the likes of `[%only %ship]` and `[%only %message-pub]` and lose none of the functionality we've looked at. The `[%only %ship]` curb just declares that the state is exclusively a `ship`. However, the `[%pro ship]` curb says that the state can be any type *which can be converted to a `ship` through an available `/con` file*. This has implications for interoperability and state transitions we have not yet fully explored.
+
+#### form:neo
+When `/imp/dm` is first created with a `%make` card, it needs to be created with some pre-defined state. The intial state it accepts has to be a `%dm-diff`. This is a little unusual, but it's essentially the same as a Gall agent sending a poke to itself `+on-init`, except the poke is defined by the external service provider. Since the service provider could be on our ship or someone else's, the `+init` arm takes both `%initiate` and `%invited` pokes. The `+init` arm is the only context in which these pokes are handles.
```

Comment:
"handles" -> "handled"

Reply:
"Since the service provider could be on our ship or someone else's, the `+init` arm takes both `%initiate` and `%invited` pokes." <- leave this sentence out

Reply:
"The `+init` arm is the only context in which these pokes are handles." allude to blasphemous design pattern here; maybe these two pokes should have a different stud, maybe init shouldn't be taking in things that aren't of this imp's state type at all

Reply:
**Key point to draw attention to** is the chicken-and-egg problem where we need a higher-level thing (aka "service provider" aka messenger) in order to start a DM with someone, because their DM doesn't exist yet and needs to be spawned, and its sub can only be created after our pub has been, etc